tasks/docker-engine.yml

```diff
@@ -19,19 +19,60 @@
     repo: 'deb https://apt.dockerproject.org/repo ubuntu-{{ ansible_distribution_release }} main'
     update_cache: yes
 
+- name: see if docker is already installed
+  stat: path=/usr/bin/docker get_checksum=False
+  register: docker_exec_stat_result
+  always_run: True 
 
-# While wildcarding the version is cute, in practice, there only seems to be one
-# version of Docker available at a time. I'm not sure what will happen on the next
-# release. Alternatively, we could install LXC Docker, which does have many more
-# versions, but not the most recent versions.
+- name: Determine installed version
```

Comment:
Sorry for being late to the party with this.  I can't tell if this is a net win since it only works on docker >= 1.8, but
```docker version --format '{{.Server.Version}}'```
returns just the version, like '1.10'.  On older versions, it fails, which I guess means you end up falling back to what you've got here...